lib/virtual-net/src/loopback.rs

```diff
@@ -0,0 +1,240 @@
+use std::collections::VecDeque;
+use std::net::SocketAddr;
+use std::sync::Mutex;
+use std::task::{Context, Poll, Waker};
+use std::{collections::HashMap, sync::Arc};
+
+use crate::tcp_pair::TcpSocketHalf;
+use crate::{
+    InterestHandler, IpAddr, IpCidr, Ipv4Addr, Ipv6Addr, NetworkError, VirtualIoSource,
+    VirtualNetworking, VirtualTcpListener, VirtualTcpSocket,
+};
+use derivative::Derivative;
+use virtual_mio::InterestType;
+
+const DEFAULT_MAX_BUFFER_SIZE: usize = 1_048_576;
+
+#[derive(Debug, Default)]
+struct LoopbackNetworkingState {
+    tcp_listeners: HashMap<SocketAddr, LoopbackTcpListener>,
+    ip_addresses: Vec<IpCidr>,
+}
+
+#[derive(Debug, Clone)]
+pub struct LoopbackNetworking {
+    state: Arc<Mutex<LoopbackNetworkingState>>,
+}
+
+impl LoopbackNetworking {
```

Comment:
Yay!

Reply:
indeed this is good news for the browser as well